### ISSUE TITLE
Filter: Deprecate component

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#10187]
**monday.com sync:** #9445817113

### Description

Once https://github.com/Esri/calcite-design-system/issues/10186 and https://github.com/Esri/calcite-design-system/issues/10044 are fulfilled, we should deprecate the `filter` component

### Proposed Advantages

Ease of use for developer implementations and support for typeahead functionality.

### Which Component

`filter`

### Relevant Info

- https://github.com/Esri/calcite-design-system/issues/10044
- https://github.com/Esri/calcite-design-system/issues/10186

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

**Comment by geospatialem** [comment]
Blocked by #10186 and #10044

**Comment by isaacbraun** [comment]
Closing as duplicate of #8877.
